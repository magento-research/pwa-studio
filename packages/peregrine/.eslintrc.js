const config = {
    parser: 'babel-eslint',
    extends: ['@magento'],
    rules: {
        "no-undef": "off"
    }
};

module.exports = config;
