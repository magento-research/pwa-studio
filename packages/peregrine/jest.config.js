module.exports = {
    displayName: 'Peregrine',
    setupFiles: ['<rootDir>/scripts/shim.js', '<rootDir>/scripts/fetch-mock.js']
};
