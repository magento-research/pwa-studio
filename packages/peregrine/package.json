{
  "name": "@magento/peregrine",
  "version": "3.0.0",
  "publishConfig": {
    "access": "public"
  },
  "description": "The core runtime of Magento PWA",
  "main": "lib/index.js",
  "scripts": {
    "build": " ",
    "build:ci": " ",
    "build:cjs": " ",
    "build:dev": " ",
    "clean": " ",
    "storybook": "start-storybook -p 9001 -c .storybook",
    "storybook:build": "build-storybook -c .storybook -o storybook-dist",
    "watch": " "
  },
  "repository": "github:magento-research/pwa-studio",
  "author": "Magento Commerce",
  "license": "(OSL-3.0 OR AFL-3.0)",
  "bugs": {
    "url": "https://github.com/magento-research/pwa-studio/issues"
  },
  "homepage": "https://github.com/magento-research/pwa-studio/tree/master/packages/peregrine#readme",
  "devDependencies": {
    "@storybook/react": "~4.1.13",
    "intl": "~1.2.5",
    "react": "~16.8.3",
    "react-apollo": "~2.5.3",
    "react-dom": "~16.8.3",
    "react-router-dom": "~5.0.0",
    "react-test-renderer": "~16.8.6",
    "rimraf": "~2.6.3"
  },
  "peerDependencies": {
    "@babel/runtime": "^7.4.2",
    "react": "^16.8.0",
    "react-apollo": "^2.5.3",
    "react-dom": "^16.8.0",
    "react-router-dom": "~5.0.0"
  },
  "engines": {
    "node": "8.10.0 || >=10.7.0",
    "yarn": ">=1.12.0"
  },
  "module": "lib/index.js",
  "jsnext:main": "lib/index.js",
  "es2015": "lib/index.js",
  "sideEffects": false
}
