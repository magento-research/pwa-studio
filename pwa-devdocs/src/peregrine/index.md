---
title: Peregrine
---

The Peregrine project is a collection of UI Components for Magento PWA projects.
Use, extend, or remix these components to create a unique Magento PWA storefront.

Visit the [`Peregrine`] repository to contribute to this project.

[`Peregrine`]: https://github.com/magento-research/pwa-studio/tree/master/packages/peregrine
