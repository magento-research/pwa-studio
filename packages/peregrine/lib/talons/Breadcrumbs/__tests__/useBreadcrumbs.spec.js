import React from 'react';
import { createTestInstance } from '@magento/peregrine';

import { act } from 'react-test-renderer';
import { useQuery } from '@apollo/client';

import { useBreadcrumbs } from '../useBreadcrumbs';

jest.mock('@apollo/client', () => ({
    useQuery: jest.fn().mockReturnValue({
        data: {
            category: {
                id: null,
                name: 'Tiki',
                url_path: 'tiki',
                url_suffix: '.html',
                breadcrumbs: [
                    {
                        category_id: 10,
                        category_name: 'Shopee',
                        category_level: 2,
                        category_url_path: 'tiki/shopee'
                    },
                    {
                        category_id: 12,
                        category_name: 'Shopee',
                        category_level: 1,
                        category_url_path: 'tiki/shopee'
                    }
                ]
            }
        },
        error: null,
        loading: false
    })
}));

const props = {
    categoryId: 1,
    query: {}
};

const log = jest.fn();

const Component = props => {
    const talonProps = useBreadcrumbs(props);
    log(talonProps);

    return <i />;
};

test('return correct shape while data is loading', () => {
    useQuery.mockReturnValueOnce({
        loading: true
    });

    const tree = createTestInstance(<Component {...props} />);
    const { root } = tree;
    const { talonProps } = root.findByType('i').props;

    expect(talonProps).toMatchSnapshot();
});


test('returns sorted data', () => {
    useQuery.mockReturnValueOnce({
        data: {
            category: {
                id: null,
                name: 'Tiki',
                url_path: 'tiki',
                url_suffix: '.html',
                breadcrumbs: [
                    {
                        category_id: 12,
                        category_name: 'Shopee',
                        category_level: 1,
                        category_url_path: 'tiki/shopee'
                    },
                    {
                        category_id: 10,
                        category_name: 'Shopee',
                        category_level: 2,
                        category_url_path: 'tiki/shopee'
                    }
                ]
            }
        },
        error: false,
        loading: false
    });
    const tree = createTestInstance(<Component {...props} />);
    const { root } = tree;
    const { talonProps } = root.findByType('i').props;
    expect(talonProps).toMatchSnapshot();
});

test('returns the correct shape', () => {
    // Act.
    createTestInstance(<Component {...props} />);

    // Assert.
    const talonProps = log.mock.calls[0][0];
    const expectedProperties = [
        'currentCategory',
        'currentCategoryPath',
        'isLoading',
        'hasError',
        'normalizedData'
    ];
    const actualProperties = Object.keys(talonProps);
    expect(actualProperties.sort()).toEqual(expectedProperties.sort());
});
