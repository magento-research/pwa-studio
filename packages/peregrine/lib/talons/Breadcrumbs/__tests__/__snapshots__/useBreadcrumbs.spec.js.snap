// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`return correct shape while data is loading 1`] = `undefined`;

exports[`returns sorted data 1`] = `undefined`;
